src/typeimpl/org/apache/xmlbeans/impl/schema/SchemaTypeLoaderBase.java

```diff
@@ -270,7 +277,27 @@ else if (! options.hasOption(XmlOptions.DOCUMENT_SOURCE_NAME))
             options = new XmlOptions( options );
             options.put( XmlOptions.DOCUMENT_SOURCE_NAME, url.toString() );
         }
+        
+        if (url.toString().startsWith("http") && retriever != null)
```

Comment:
Use http:// and https:// as you did in mule, otherwise a file named httpFileName.xsd won't be correctly handled.